server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -250,4 +285,12 @@ class LivyConf(loadDefaults: Boolean) extends ClientConf[LivyConf](null) {
     Option(get(entry)).map(_.split("[, ]+").toSeq).getOrElse(Nil)
   }
 
+  override def getConfigsWithAlternatives: JMap[String, DeprecatedConf] = {
+    JCollections.unmodifiableMap(configsWithAlternatives.asJava)
```

Comment:
you are correct, I misread a scala article and though `asJava` would make it mutable, but it correctly keeps the scala (un)mutability.